lib/classes/Service.js

```diff
@@ -149,16 +149,27 @@ class Service {
         // setup function.name property
         const stageNameForFunction = options.stage || this.provider.stage;
         _.forEach(that.functions, (functionObj, functionName) => {
+          const targetFunction = that.functions[functionName];
           if (!functionObj.events) {
-            that.functions[functionName].events = [];
+            targetFunction.events = [];
           }
           if (!_.isArray(functionObj.events)) {
             throw new SError(`Events for "${functionName}" must be an array,` +
                              ` not an ${typeof functionObj.events}`);
           }
 
+          // Translate HTTP events written in string notation to object notation
+          _.forEach(functionObj.events, (event, index) => {
```

Comment:
Here's how I went about it (feel free to comment/suggest changes/ask me to undo/....):

I introduced a new "synthetic" event called "_system_:normalize". The "_system_" prefix is intended to designate the fact that this is an internal event (as opposed to an event that is derived from the command line passed to the program). The deploy/aws plugin provides a hook (on this event) which does the normalization.